### PR TITLE
GBE 398

### PR DESCRIPTION
**Pull request by bethlakshmi**
Basic integration is done.
Needs more integration testing - key tests coming tomorrow via tickets, but now is a good time for a pull request and review.

### CODE REVIEW

File: expo/gbe/menu.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from cms.menu_bases import CMSAttachMenu
+from menus.base import Menu, NavigationNode
+from menus.menu_pool import menu_pool
+
+from gbe.functions import *
+from gbe.models import *
+
+'''
+  This is the best simulation of our old login menu I could come up with - the Django Menu
+  doesn't offer Clickable top level menu items if there are children items.
```
Comment (jonkiparsky) [inline expo/gbe/menu.py]: This is a Good Thing. Having the top level menu items clickable is a bug, not a feature.

File: expo/gbe/templates/gbe/incl_lp_left.tmpl [hunk 2]
```diff
@@ -11,7 +11,7 @@ shows your performing in, or shifts you've volunteered for) and all sorts of oth
 cool stuff.
   {% if  user.profile.preferences.show_hotel_infobox %}
 <br><hr><br>
-Have you reserved your room at The Hyatt Regency Hotel yet? A single or double room is just $149 per night until January 19, 2015. Three or four people in a room are slighly more expensive. The hotel sold out last year! Don't get left out in the cold!. Make or manage your reservation <a href = "https://aws.passkey.com/event/11159123/owner/371/home">on-line</a> or by calling the hotel at (617) 492-1234 and asking for "The Great Burlesque Exposition" rate.
+Have you reserved your room at The Hyatt Regency Hotel yet? A single or double room is just $149 per night until January 19, 2015. Three or four people in a room are slighly more expensive. The hotel snew out last year! Don't get left out in the cnew!. Make or manage your reservation <a href = "https://aws.passkey.com/event/11159123/owner/371/home">on-line</a> or by calling the hotel at (617) 492-1234 and asking for "The Great Burlesque Exposition" rate.
```
Comment (jonkiparsky) [inline expo/gbe/templates/gbe/incl_lp_left.tmpl]: Ha! I see your auto-replace... :)

File: expo/gbe/templates/gbe/fashionfair.tmpl [hunk 3]
```diff
@@ -7,7 +7,7 @@
...
+   <B>The Great Burlesque Exposition</b> offers two different (but equally fabulous!) types of shopping experience! First, is our <I>Vintage Fashion Fair</I>, located on the first floor in the President's Ballroom, next door to the theater.  Open on Saturday and Sunday, from noon until 8:00 p.m., it's home to dozens of delightful vendors selling everything you need, whether you're a professional performer or just want to look spectacular!<P><I>The Vintage Fashion Fair</I> is also home to our <B>Costume Display</b>, our <B>Art Exhibit</b>, and our <B>Caf&eacute;</B>.<P>Feeling peckish between classes? Want to grab a quick snack or beverage? No need to run out into the cnew or even up to the hotel restaurant! Attendees of <I>The Great Burlesque Exposition</I> can fortify themselves with soups, wraps, fresh fruit, and more... all at very reasonable prices!
```
Comment (jonkiparsky) [inline expo/gbe/templates/gbe/fashionfair.tmpl]: Auto-replace...

File: expo/gbe/templates/gbe/update_profile.tmpl [hunk 4]
```diff
@@ -9,7 +9,7 @@
 <span class='header'><h3>{%if display_name%}{{display_name}}'s{%else%}Your{%endif%} Profile</h3></span><br>
 
 Your privacy is very important to us. We promise that your information
-will not be sold or traded. That said, we need your complete and
+will not be snew or traded. That said, we need your complete and
```
Comment (jonkiparsky) [inline expo/gbe/templates/gbe/update_profile.tmpl]: Autoreplace... please, let's don't use copy/paste tools on code!

File: expo/templates/admin/delete_confirmation.html [hunk 5]
```diff
@@ -1,42 +1,43 @@
...
-    {% endif %}
-    {% if protected %}
-        <p>{% blocktrans with escaped_object=object %}Deleting the {{ object_name }} '{{ escaped_object }}' would require deleting the following protected related objects:{% endblocktrans %}</p>
-        <ul>
-        {% for obj in protected %}
-            <li>{{ obj }}</li>
-        {% endfor %}
-        </ul>
-    {% endif %}
+{% if perms_lacking %}
```
Comment (jonkiparsky) [inline expo/templates/admin/delete_confirmation.html]: Why are we losing this indentation? Shouldn't form reflect structure?

File: expo/templates/admin/delete_selected_confirmation.html [hunk 6]
```diff
@@ -1,47 +1,48 @@
...
-    {% endif %}
-    {% if protected %}
-        <p>{% blocktrans %}Deleting the selected {{ objects_name }} would require deleting the following protected related objects:{% endblocktrans %}</p>
-        <ul>
-        {% for obj in protected %}
-            <li>{{ obj }}</li>
-        {% endfor %}
-        </ul>
-    {% endif %}
+{% if perms_lacking %}
```
Comment (jonkiparsky) [inline expo/templates/admin/delete_selected_confirmation.html]: Same as previous - don't know why we're flattening this

File: expo/templates/admin/delete_selected_confirmation.html [hunk 7]
```diff
@@ -1,47 +1,48 @@
...
-    <div>
-    {% for obj in queryset %}
-    <input type="hidden" name="{{ action_checkbox_name }}" value="{{ obj.pk|unlocalize }}" />
-    {% endfor %}
-    <input type="hidden" name="action" value="delete_selected" />
-    <input type="hidden" name="post" value="yes" />
-    <input type="submit" value="{% trans "Yes, I'm sure" %}" />
-    </div>
-    </form>
+<p>{% blocktrans %}Are you sure you want to delete the selected {{ objects_name }}? All of the following objects and their related items will be deleted:{% endblocktrans %}</p>
```
Comment (jonkiparsky) [inline expo/templates/admin/delete_selected_confirmation.html]: Again, same

**Comment by jonkiparsky** [comment]
Overall, looks good. A few questionable changes, noted already. 
Will try to stand this up on my local machine in the AM and do more thorough review.

**Comment by jonkiparsky** [comment]
It looks like merging in the pep-8 changes won't be too hard, but we'll
have to be a little careful. I suspect it'll be easier to do pep-8 first,
then this. (That way, your changes lie on top of mine, and if there's
lossage it'll just be to the pep-8 stuff)

**Comment by bethlakshmi** [comment]
the two recent commits should fix the auto replace and indentation problems with comments on the check in.  Don't care what the order of operations on pep-8 vs. this - your approach works for me.

To stand up locally, you'll want to follow the instructions here:
http://docs.django-cms.org/en/3.0.13/how_to/install.html?highlight=djangocms_admin_style

And reference requirements.txt checked in at the highest level of gbe2 - I found that their example requirements.txt actually specified versions that DON'T work because they are presuming Django1.7 and we're using 1.6.5.  The requirements.txt is largely what worked for me - but I didn't necessarily run it with a totally clean machine, so I don't feel 100% certain of it.

You also may need to make a media directory and static directory in places you didn't have them before.  I found that I *had* to have a static directory in expo/expo although nothing is ever put there (it's all in the actual STATIC_ROOT).

**Pull request merged**